src/api/Metadata.js

```diff
@@ -137,14 +136,15 @@ class Metadata extends File {
      * @param {string} scope - metadata scope
      * @return {Object} array of metadata templates
      */
-    async getTemplates(id: string, scope: string): Promise<Array<MetadataEditorTemplate>> {
+    async getTemplates(id: string, scope: string, isInstanceId?: boolean): Promise<Array<MetadataEditorTemplate>> {
+        this.errorCode = ERROR_CODE_FETCH_METADATA_TEMPLATES;
         let templates = {};
         try {
             templates = await this.xhr.get({
```

Comment:
I assume this API is paginated. Can you use one of the existing API methods to fetch all the metadata to make it more future proof?

Reply:
Since templates are limited to 500, we can hold off doing the change for now, might involve a larger change for no benefit.

Reply:
The change is trivial to implement. You just need to extend the correct api